get_smb_mount.go

```diff
@@ -0,0 +1,101 @@
+package getter
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+// SmbMountGetter is a Getter implementation that will download a module from
+// a shared folder using the file system using FileGetter implementation.
+// For Unix and MacOS users, the Getter will look for usual system specific mount paths such as:
+// /Volumes/ for MacOS
+// /run/user/1000/gvfs/smb-share:server=<hostIP>,share=<path> for Unix
+type SmbMountGetter struct{}
+
+func (g *SmbMountGetter) Mode(ctx context.Context, u *url.URL) (Mode, error) {
+	if u.Host == "" || u.Path == "" {
+		return 0, new(smbPathError)
+	}
+
+	prefix, path := g.findPrefixAndPath(u)
+	u.Path = prefix + path
+
+	return new(FileGetter).Mode(ctx, u)
+}
+
+func (g *SmbMountGetter) Get(ctx context.Context, req *Request) error {
+	if req.u.Host == "" || req.u.Path == "" {
+		return new(smbPathError)
+	}
+
+	prefix, path := g.findPrefixAndPath(req.u)
+	req.u.Path = prefix + path
+
+	return new(FileGetter).Get(ctx, req)
+}
+
+func (g *SmbMountGetter) GetFile(ctx context.Context, req *Request) error {
+	if req.u.Host == "" || req.u.Path == "" {
+		return new(smbPathError)
+	}
+
+	prefix, path := g.findPrefixAndPath(req.u)
+	req.u.Path = prefix + path
+
+	return new(FileGetter).GetFile(ctx, req)
+}
+
+func (g *SmbMountGetter) findPrefixAndPath(u *url.URL) (string, string) {
+	var prefix, path string
+	switch runtime.GOOS {
+	case "windows":
+		prefix = string(os.PathSeparator) + string(os.PathSeparator)
+		path = filepath.Join(u.Host, u.Path)
+	case "darwin":
+		prefix = string(os.PathSeparator)
+		path = filepath.Join("Volumes", u.Path)
+	}
+	return prefix, path
+}
+
+func (g *SmbMountGetter) Detect(req *Request) (bool, error) {
+	if runtime.GOOS == "linux" {
+		// Linux users should use smbclient command.
```

Comment:
Can you please explain why in the comment ( In case the why gets fixed in the future )

Reply:
Do you mean why to use smbclient command? Like for example, because it is safer?